File: testdata/bench/go/BenchmarkTest00009.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via string builder pattern
func Handler00009(w http.ResponseWriter, r *http.Request) {
	col := r.FormValue("column")
	val := r.FormValue("value")
	query := "SELECT * FROM products WHERE " + col + " = '" + val + "'"
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Exec(query)
}

File: testdata/bench/kotlin/BenchmarkTest00009.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via cookie value
suspend fun handler00009(call: ApplicationCall) {
    val role = call.request.cookies["role"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    conn.createStatement().executeQuery("SELECT * FROM users WHERE role = '${role}'")
    call.respondText("OK")
}
